src/lib.rs

```diff
@@ -0,0 +1 @@
+mod sg;
```

Comment:
Cli can potentially live inside the `main.rs` if the clap library takes care of the functionality we need. Then inside `main.rs` we can init an `App` which would come from `sg::App`